packages/samples/sudt/src/actors/sudt.model.ts

```diff
@@ -0,0 +1,152 @@
+/**
+ * @module src/actors/omnilock.model
+ * @description
+ * This is the actor model for omnilock, which is used to gather omnilock cells to generate record models.
+ */
+
+import {
+  ActorProvider,
+  Param,
+  ActorReference,
+  CellPattern,
+  JSONStore,
+  OutPointString,
+  SchemaPattern,
+  UpdateStorageValue,
+  TypeFilter,
+  Sudt,
+} from '@ckb-js/kuai-models'
+import type { Cell, HexString, Script } from '@ckb-lumos/base'
+import { number, bytes } from '@ckb-lumos/codec'
+import { InternalServerError } from 'http-errors'
+import { BI, utils, config } from '@ckb-lumos/lumos'
+import { MIN_SUDT_WITH_OMINILOCK, TX_FEE } from '../const'
+
+/**
+ * add business logic in an actor
+ */
+@ActorProvider({ ref: { name: 'sudt', path: `/:lockArgs/:typeArgs/` } })
+@TypeFilter()
+@Sudt()
```

Comment:
From the design of the `Model`, we need to use `Lock` to decorate the `SudtModel`. There must be a `lockscript` in the instance. As we only use `omnilock`, we should add the `@Omnilock()` here.

Reply:
This model will collect all the sudt cells without a lock script filter. We need to collect some locks owned by one user.

Reply:
> This model will collect all the sudt cells without a lock script filter. We need to collect some locks owned by one user.

The `Model` is designed to related `lock`, here I mark it down. https://github.com/ckb-js/kuai/issues/504#issuecomment-1756666591
Or we should change the design to support it.

Reply:
After merge https://github.com/ckb-js/kuai/pull/509. Sudt model can listen on type script without lock script.

Reply:
I have change ```@ActorProvider({ ref: { name: 'sudt', path: `/:lockArgs/:typeArgs/` } }) ``` to ```@ActorProvider({ ref: { name: 'sudt', path: `/:args/` } })```

Reply:
This has been fixed by https://github.com/ckb-js/kuai/pull/509